### ISSUE TITLE
[BUG]  MLflow enforces the system theme upon page refresh

### ISSUE THREAD

**Issue by danilopeixoto** [mlflow/mlflow#12378]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- Client: 2.14.0
- Tracking server: 2.14.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 22.04
- **Python version**: 3.12.1
- **yarn version, if running the dev UI**: N/A

### Describe the problem

MLflow enforces the system theme upon page refresh, creating a UX problem. Users are unable to choose between Light, Dark, or System themes.

**Example**

If a user with a Dark system theme selects the Light theme and refreshes the page, the UI will revert to Dark mode, contradicting the recent user choice.

### Tracking information

```shell
N/A
```

### Code to reproduce issue

```
N/A
```

### Stack trace

```
N/A
```

### Other info / logs

```
[2024-06-17 10:51:29 -0300] [482089] [INFO] Starting gunicorn 22.0.0
[2024-06-17 10:51:29 -0300] [482089] [INFO] Listening at: http://127.0.0.1:5000 (482089)
[2024-06-17 10:51:29 -0300] [482089] [INFO] Using worker: sync
[2024-06-17 10:51:29 -0300] [482090] [INFO] Booting worker with pid: 482090
[2024-06-17 10:51:29 -0300] [482091] [INFO] Booting worker with pid: 482091
[2024-06-17 10:51:29 -0300] [482092] [INFO] Booting worker with pid: 482092
[2024-06-17 10:51:29 -0300] [482093] [INFO] Booting worker with pid: 482093
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
I was able to reproduce the issue:

https://github.com/mlflow/mlflow/assets/17039389/68e8625a-9be3-4296-80f4-c19a919186b9

**Comment by daniellok-db** [comment]
Should be fixed by #12386, thanks for reporting!
